### PR TITLE
Implement the onboarding UI

### LINKED ISSUE #5: Implement generic onboarding pages
[The Figma document](https://www.figma.com/file/uLiqrlxmOB5xCppIzmiUXV/Teacher-Workout?node-id=0%3A1) contains the entire UI of the project. 

You can export any image (asset) directly from Figma. [Here's how](https://drive.google.com/file/d/1KJncuAGFvKEo6zrjC9H3O59auITE0q2h/view). 

**Please do not try to recreate the UI design exactly.** Prioritize using de default UI mobile components, and following the material design guidelines. The UI objects should be positioned roughly the same as in the Figma screens. 

[The mobile flow can be found in the project Wiki](https://github.com/code4romania/teacher-workout-backend/wiki/Teacher-Workout-Mobile-flow).

_Please define related colors and text & button styles in the corresponding values files. Please extract all text values in the corresponding xml files so we can easily adjust them. <3_

---

After having just created their account and after logging in, a user is redirected to the on-boarding pages. They can either go through the entire on-boarding process or skip it all together. If they do not skip nor complete the on-boarding process, they can't access any other screen in the app. 

We need these 3 generic on-boarding pages, which will later be populated by text describing the main functionalities of the app. 

image

### PR DESCRIPTION
**Pull request by lukstbit**
Closes #5  

This PR implements the UI from the Figma mockup. The images from the mockup don't look too good because I imported them as .png.

### How has it been tested?

Manually, as a helper I added a button to navigate directly to onboarding from the login screen(It was also added to the PR).

### CODE REVIEW

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt [hunk 1]
```diff
@@ -66,6 +66,8 @@ object Dependencies {
 
             // Integration with observables
             const val LIVEDATA = "androidx.compose.runtime:runtime-livedata:$VERSION"
+
+            const val PAGER = "com.google.accompanist:accompanist-pager:0.20.0"
```
Comment (AlexandraDamaschin) [inline buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: It might be better to extract the version in a variable?

File: features/account/build.gradle.kts [hunk 2]
```diff
@@ -10,4 +11,12 @@ android {
 
 dependencies {
     implementation(project(":commons:ui"))
+    implementation(Dependencies.AndroidX.Compose.PAGER)
+}
+
+// needed to build the app because the compose pager is currently marked as experimental
```
Comment (AlexandraDamaschin) [inline features/account/build.gradle.kts]: Maybe we can add a todo for this to delete it after the pager library gets stable?

File: features/account/src/main/kotlin/com/teacherworkout/features/account/AccountFeature.kt [hunk 3]
```diff
@@ -33,6 +34,12 @@ fun NavGraphBuilder.accountFeature(navHostController: NavHostController) {
         composable(AppDestinations.reset_password) {
             ResetPasswordScreenDestination(navHostController)
         }
+
+        composable("onboarding") {
+            OnBoardingScreen {
+                // navigate to main here
```
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/AccountFeature.kt]: Wondering how we could connect this to navigate to home screen. Would there be a circular dependency again? Should we refactor this in order to be able to implement this?

File: features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthScreen.kt [hunk 4]
```diff
@@ -82,6 +87,13 @@ fun AuthScreen(
                     }
                 }
             }
+            // TODO remove this after onboarding is fully implemented
+            if (BuildConfig.DEBUG) {
+                Spacer(modifier = Modifier.height(space16dp))
+                Button(onClick = { navController.navigate("onboarding") }) {
```
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthScreen.kt]: Can't we implement that after you press the login to be redirected to this? Always for now and leave a todo to refactor it after we connect it to the backend.

File: features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/GenericOnBoardingStep.kt [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            style = TextStyle(
+                color = colorResource(id = R.color.onboarding_text_description),
+                fontSize = MaterialTheme.typography.body2.fontSize,
+            ),
+            modifier = Modifier.padding(horizontal = space16dp)
+        )
+    }
+}
+
+@Preview
```
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/GenericOnBoardingStep.kt:88]: Is considered a good practice to leave the previews of the screen in the class or those should be kept only for development purposes?

File: features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/NextStepButton.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.teacherworkout.features.account.R
+
+@Composable
+fun NextStepButton(hasMoreSteps: Boolean, onNext: () -> Unit, modifier: Modifier = Modifier) {
```
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/NextStepButton.kt:26]: Could we implement that the onNext button, when it's at the final onboarding screen, to redirect to the home screen?

File: features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+                .padding(all = space16dp)
+                .fillMaxWidth(),
+        ) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.SpaceBetween,
+                modifier = Modifier.fillMaxWidth(),
+            ) {
+                TextButton(
+                    onClick = onNavigateToMain
```
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt:61]: Could we implement the navigation to the main screen as well? Or we will do this in another PR?

File: features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+                ) {
+                    Text(
```
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt:63]: In the design, the skip button it's placed in the right top corner, could we add it over the image in the right top corner? 
image

File: features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+                            localScope.launch {
+                                pagerState.animateScrollToPage(pagerState.currentPage + 1)
+                            }
+                        } else {
+                            onNavigateToMain()
+                        }
+                    }
+                )
+            }
+            PageIndicator(
```
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt:83]: Could we add the page indicator over the image? 
image

File: features/account/src/main/res/values-ro-rRO/strings.xml [hunk 10]
```diff
@@ -37,4 +37,10 @@
     <string name="reset_password_loading_label">Resetează</string>
     <string name="reset_password_failure_label">Nu v-am putut reseta parola. Vă rugăm să încercați din nou!</string>
     <string name="reset_password_success_label">Un e-mail pentru resetarea parolei a fost trimis la adresa de e-mail introdusă.</string>
+
+    <!-- OnBoarding screen related strings -->
```
Comment (AlexandraDamaschin) [inline features/account/src/main/res/values-ro-rRO/strings.xml:41]: Thanks for adding the translations to Romanian :+1:

**Review by AlexandraDamaschin — COMMENTED** [review]
Some questions about this and what we can implement in this PR or in future PRs

File: features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt (hunk 8 above)
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt:63]: I've placed the "skip" button at the bottom because I thought this is a more natural position for the button. This is where the button is present in the on boarding of many apps, when setting your google account etc. The same thing for the pager indicator.

If we should fully replicate the mockup I can move the button, of course.

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt (hunk 1 above)
Comment (lukstbit) [reply buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: Done!

File: features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/GenericOnBoardingStep.kt (hunk 5 above)
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/GenericOnBoardingStep.kt:88]: I don't know if it's a good practice or not, I declared them in the same class for easier access :)
My understanding is they will be automatically removed either by the tooling or being dead code which isn't referenced anywhere.

File: features/account/build.gradle.kts (hunk 2 above)
Comment (lukstbit) [reply features/account/build.gradle.kts]: Done! I definitely should have done that from the start!

File: features/account/src/main/kotlin/com/teacherworkout/features/account/AccountFeature.kt (hunk 3 above)
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/AccountFeature.kt]: We can simply navigate to home by its destination with no issues. I added the actual code to do this in my fix commit.

File: features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthScreen.kt (hunk 4 above)
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthScreen.kt]: Done! It makes more sense as on boarding is part of the actual navigation after login(if it wasn't already done by the user).

File: features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt (hunk 7 above)
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt:61]: On both clicking the skip and the last finish button the app will now navigate to the home screen(landing at the moment).

File: features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/NextStepButton.kt (hunk 6 above)
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/NextStepButton.kt:26]: Done!

File: features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt (hunk 8 above)
Comment (AlexandraDamaschin) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/onboarding/OnBoardingScreen.kt:63]: You're right, I've looked over and this seems to be the most used way for android. @catileptic could you help us here? Should we keep the design or we are good as it's implemented now
Screenshot_20211027-160343
?

**Comment by AlexandraDamaschin** [comment]
After we get the ok regarding the "Skip" button, we are good to merge this ticket from my point of view :) 
Thanks for all the help :100:

**Comment by catileptic** [comment]
I second @AlexandraDamaschin's decision to go with this placement of the "skip" button

**Review by AlexandraDamaschin — APPROVED** [review]
Does this improve software quality? Yes
